homeassistant/components/esphome/__init__.py

```diff
@@ -705,6 +705,11 @@ def __init__(
         self._component_key = component_key
         self._key = key
         self._state_type = state_type
+        if entry_data.device_info is not None:
+            self._attr_has_entity_name = entry_data.device_info.friendly_name not in (
+                None,
+                "",
+            )
```

Comment:
```suggestion
        if entry_data.device_info is not None and entry_data.device_info.friendly_name:
            self._attr_has_entity_name = entry_data.device_info.friendly_name
```

Is a truth check ok here?

Reply:
Makes sense